src/mbgl/layout/symbol_layout.cpp

```diff
@@ -229,7 +232,8 @@ void SymbolLayout::prepare(uintptr_t tileUID,
     const bool textAlongLine = layout.get<TextRotationAlignment>() == AlignmentType::Map &&
         layout.get<SymbolPlacement>() == SymbolPlacementType::Line;
 
-    for (const auto& feature : features) {
+    for (auto it = features.begin(); it != features.end(); ++it) {
+        auto feature = *it;
```

Comment:
This is creating a copy; it should be `auto& feature = *it;`. Then `feature.geometry.clear();` will have a meaningful effect.